userprofile.py

```diff
@@ -13,7 +13,7 @@ class UserProfile:
 
 	_id: Optional[ObjectId] = None
 	user_id: int = 0
-	pokemon: list[ObjectId] = []
+	pokemon: set[ObjectId] = set()
```

Comment:
I found out today that this actually creates a shared reference to the value being assigned here for all the instances of this class that are created. So lets go ahead and change that here
```suggestion
	pokemon: set[ObjectId]
	
	def __init__(self):
		self.pokemon = set()
```